### DISCUSSION TITLE
Case insensitive querying in YesSQL

### DISCUSSION THREAD

**Discussion by scleaver** [OrchardCMS/OrchardCore#13293] [Q&A]
There doesn't seem to be a way to do case-insensitive text comparison in YesSQL - can someone shed some light?

So for example all these will throw the error "Operation is not valid due to the current state of the object.":
`c.DisplayText.Contains(name, StringComparison.InvariantCultureIgnoreCase)`
or 
`c.DisplayText.ToLower() == name.ToLower()`
or
`c.DisplayText.Equals(name, StringComparison.InvariantCultureIgnoreCase)`

Changing the query back to this resolves the issue:
`c.DisplayText == name`

**Comment by jtkech** [comment]
From memory logic operators in predicates are mapped to the related sql commands (for example `is like`) that are case insensitive, idem for some well known lookup methods as `.Contains()`, but without the `StringComparison`.
